File: Algorithms/Sort/Selection Sort/src.java
/*
ALGORITHM : Selection Sort
WORST CASE => {
    PERFORMANCE:  O(n^2)
    SPACE:        O(1)
}
*/

import java.util.Arrays;

public class SelectionSort {
    public static void main(String[] args) {
        int[] arr;
        arr = new int[] {6,5,4,3,2,1};

        int l = arr.length;
        int minIndex, temp;
        for(int i=0;i<l;i++) {
            minIndex = l-1;
            for(int j=i;j<l;j++) {
                if(arr[minIndex] > arr[j]) {
                    minIndex = j;
                }
            }

            temp = arr[minIndex];
            arr[minIndex] = arr[i];
            arr[i] = temp;
        }
        System.out.print(Arrays.toString(arr));
    }
}

File: Algorithms/Sort/Selection Sort/src.kt
/*
ALGORITHM : Selection Sort
WORST CASE => {
    PERFORMANCE:  O(n^2)
    SPACE:        O(1)
}
*/

fun main() {
    val arr = arrayOf(9, 8, 7, 6, 5, 4, 3, 2, 1)
    val l = arr.size - 1
    var minIndex:Int
    var tmp:Int
    for(i in 0..l) {
        minIndex = i
        for(j in i..l) {
            if(arr[minIndex] > arr[j]) {
                minIndex = j
            }
        }

        tmp = arr[minIndex]
        arr[minIndex] = arr[i]
        arr[i] = tmp
    }

    print("Sorted Array: ")
    println(arr.contentToString())
}